nilearn_sandbox/datasets.py

```diff
@@ -0,0 +1,70 @@
+import os
+import cookielib
+import getpass
+
+from sklearn.datasets.base import Bunch
+
+from nilearn import datasets
+from nilearn._utils.compat import _urllib
+
+from ._utils.compat import _input
+
+
+def _get_nitrc_auth_cookie(login=None):
+    if login is None:
+        login = _input('login: ')
+    passwd = getpass.getpass('password: ')
```

Comment:
Sure, I don't advocate that either. 

My point is that you're forcing a particular type of password input deep in the code; I think it's more appropriate to accept it as a parameter here, and make the decision of how to input it at a higher level (like within the example script).